pkg/reconciler/v1alpha1/revision/reconcile_resources.go

```diff
@@ -45,51 +46,12 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 	deploymentName := resourcenames.Deployment(rev)
 	logger := logging.FromContext(ctx).With(zap.String(logkey.Deployment, deploymentName))
 
-	deployment, err := c.deploymentLister.Deployments(ns).Get(deploymentName)
-	if apierrs.IsNotFound(err) {
-		// Deployment does not exist. Create it.
-		rev.Status.MarkDeploying("Deploying")
-		deployment, err = c.createDeployment(ctx, rev)
-		if err != nil {
-			logger.Errorf("Error creating deployment %q: %v", deploymentName, err)
-			return err
-		}
-		logger.Infof("Created deployment %q", deploymentName)
-	} else if err != nil {
-		logger.Errorf("Error reconciling deployment %q: %v", deploymentName, err)
+	deployment, err := c.deployment(ctx, rev, deploymentName, logger)
+	if err != nil {
 		return err
-	} else if !metav1.IsControlledBy(deployment, rev) {
-		// Surface an error in the revision's status, and return an error.
-		rev.Status.MarkResourceNotOwned("Deployment", deploymentName)
-		return fmt.Errorf("Revision: %q does not own Deployment: %q", rev.Name, deploymentName)
-	} else {
-		// The deployment exists, but make sure that it has the shape that we expect.
-		deployment, err = c.checkAndUpdateDeployment(ctx, rev, deployment)
-		if err != nil {
-			logger.Errorf("Error updating deployment %q: %v", deploymentName, err)
-			return err
-		}
 	}
```

Comment:
I'd leave everything above this line alone, it's a common pattern across virtually every reconcile we have.  If I were to encapsulate anything, it'd be what's below this line (determination of status based on "deployment").

Reply:
Sent: https://github.com/knative/serving/pull/3589

Reply:
I'll wait till #3589 is merged in, but I still believe deployment changing within a control flow statement is surprising when reading `reconcileDeployment`. It's easier to scan the function when variables are only assigned once and can be treated as constants. So I believe it's still best to separate out the `deployment` function which has the purpose of returning a `Deployment` object.

Reply:
My concern is consistency more than dogmatically adhering to the shape we have now.  If we're going to change it here, we should change it ~everywhere.  I'd probably call this something like `createOrUpdateDeployment` (that doesn't capture "get"), but it is clearer than `deployment()` :(